packages/ai/src/ui/chat-store.ts

```diff
@@ -419,7 +418,7 @@ export class ChatStore<
     id: string,
   ): Chat<MESSAGE_METADATA, InferUIDataParts<UI_DATA_PART_SCHEMAS>> {
     if (!this.hasChat(id)) {
-      throw new Error(`chat '${id}' not found`);
+      this.addChat(id, []);
     }
```

Comment:
@lgrammel we decided to add this logic to the chat store itself! this resolves a concern w/ the svelte approach where upon switching chats after `Chat` class is initialized, new chat id's did not exist in the store